### PR TITLE
add plurals to .POT

### PR DESCRIPTION
**Pull request by iuriegutu95**
Adding plural suport for .POT export. The compiler was already working. I added a structure to report the request from plural in code to the extracting function.

### CODE REVIEW

**Comment by iuriegutu95** [comment]
Guys, this has been updated to fix importing of POT as well. @marcglasberg Can you merge this?

**Comment by marcglasberg** [comment]
@bauerj Hello Johann, could you please approve, or not, this PR?

File: lib/i18n_getstrings.dart [hunk 1]
```diff
@@ -2,24 +2,63 @@ import 'dart:io';
...
+  final String string;
+  final bool pluralRequired;
+
+  ExtractedString(this.string, {this.pluralRequired = false});
+
+  @override
+  List<Object> get props => [string, pluralRequired];
+}
+
+class DecodedSintax {
```
Comment (bauerj) [inline lib/i18n_getstrings.dart:37]: What does this mean? Did you mean Syntax?

File: lib/i18n_getstrings.dart [hunk 2]
```diff
@@ -55,17 +93,26 @@ class StringExtractor extends UnifyingAstVisitor<void> {
...
     }
 
     return super.visitSimpleStringLiteral(node);
   }
 
   @override
   void visitAdjacentStrings(AdjacentStrings node) {
-    if (_hasI18nSyntax(node.strings.last, node.parent!)) {
-      strings.add(node.stringValue!);
+    final DecodedSintax sintax =
```
Comment (bauerj) [inline lib/i18n_getstrings.dart:109]: There is too much duplicated code now. Please factor this and the same code above out into a new method, e.g. `_handleI18nSyntax`.

File: lib/i18n_getstrings.dart [hunk 3]
```diff
@@ -2,24 +2,63 @@ import 'dart:io';
...
+    this.modifiers = const [],
+  });
+
+  DecodedSintax.valid(List<I18nRequiredModifiers> modifiers)
+      : this._(valid: true, modifiers: modifiers);
+
+  DecodedSintax.invalid() : this._(valid: false);
+
+  final bool valid;
+  final List<I18nRequiredModifiers> modifiers;
```
Comment (bauerj) [inline lib/i18n_getstrings.dart:49]: This looks a bit complicated for what you are trying to achieve. Wouldn't it be better to just add a `bool isPlural` to the class?

**Review by bauerj — CHANGES_REQUESTED** [review]
Thanks, this is awesome!

I do have a few things I would advise you to change though. Apart from those, this looks like excellent work!

**Comment by iuriegutu95** [comment]
@bauerj I've added the changes. For the 3rd point - I wouldn't agree that it's too complex, it's ready for future usage like gender addition, etc. I would leave it like it is.

**Comment by iuriegutu95** [comment]
@bauerj following my comment, leave me a feedback if you think something else needs addressing, otherwise please merge it.

**Review by bauerj — APPROVED** [review]
ship_it

**Comment by bauerj** [comment]
Thanks for your further improvements. IMO this is ready to merge now @marcglasberg.

**Pull request merged**